tool/scorecard.dart

```diff
@@ -0,0 +1,159 @@
+}
+
+void printMarkdownTable(ScoreCard scorecard) {
+  print(
+      '| name | fix | flutter user | flutter repo | pedantic | stagehand | status | bug refs |');
+  print('| :--- | :---: | :---:| :---: | :---: | :---: | :---: | :--- |');
+  scorecard.forEach((lint) {
+    var sb = StringBuffer('| `${lint.name}` |');
+    sb.write('${lint.hasFix ? " $bulb" : ""} |');
+    sb.write('${lint.ruleSets.contains('flutter') ? " $checkMark" : ""} |');
+    sb.write(
+        '${lint.ruleSets.contains('flutter_repo') ? " $checkMark" : ""} |');
+    sb.write('${lint.ruleSets.contains('pedantic') ? " $checkMark" : ""} |');
+    sb.write('${lint.ruleSets.contains('stagehand') ? " $checkMark" : ""} |');
+    sb.write('${lint.maturity != 'stable' ? ' **${lint.maturity}** ' : ""} |');
+    sb.write(' ${lint.bugReferences.join(", ")} |');
+    print(sb.toString());
+  });
+}
+
+class ScoreCard {
+  int get lintCount => registeredLints.length;
+
+  List<LintScore> scores = <LintScore>[];
+
+  void add(LintScore score) {
+    scores.add(score);
+  }
+
+  void forEach(void f(LintScore element)) {
+    scores.forEach(f);
+  }
+
+  static Future<List<String>> _getLintsWithFixes() async {
+    var client = http.Client();
+    var req = await client.get(
+        'https://raw.githubusercontent.com/dart-lang/sdk/master/pkg/analysis_server/lib/src/services/correction/fix_internal.dart');
+    var lintsWithFixes = <String>[];
+    for (var word in req.body.split(RegExp('\\s+'))) {
+      if (word.startsWith('LintNames.')) {
+        var lintName = word.substring(10);
+        if (lintName.endsWith(')')) {
+          lintName = lintName.substring(0, lintName.length - 1);
+        }
+        lintsWithFixes.add(lintName);
+      }
+    }
+    return lintsWithFixes;
+  }
+
+  static Future<List<Issue>> _getIssues() async {
+    var github = createGitHubClient();
+    var slug = RepositorySlug('dart-lang', 'linter');
+    return github.issues.listByRepo(slug).toList();
+  }
+
+  static Future<ScoreCard> calculate() async {
+    var lintsWithFixes = await _getLintsWithFixes();
+    var flutterRuleset = await flutterRules;
+    var flutterRepoRuleset = await flutterRepoRules;
+    var pedanticRuleset = await pedanticRules;
+    var stagehandRuleset = await stagehandRules;
+
+    var issues = await _getIssues();
+    var bugs = issues.where(_isBug).toList();
+
+    var scorecard = ScoreCard();
+    for (var lint in registeredLints) {
+      var ruleSets = <String>[];
+      if (flutterRuleset.contains(lint.name)) {
+        ruleSets.add('flutter');
+      }
+      if (flutterRepoRuleset.contains(lint.name)) {
+        ruleSets.add('flutter_repo');
+      }
+      if (pedanticRuleset.contains(lint.name)) {
+        ruleSets.add('pedantic');
+      }
+      if (stagehandRuleset.contains(lint.name)) {
+        ruleSets.add('stagehand');
+      }
+      var bugReferences = <String>[];
+      for (var bug in bugs) {
+        if (bug.title.contains(lint.name)) {
```

Comment:
How accurate is this heuristic? Have we been good about ensuring that issues related to a lint always have the lint name in the title?

Also, this doesn't account for the issues in the sdk issue tracker. Should we start being more careful about where lint issues are located?

Reply:
I don't know how accurate it is but it does match a few. 😄   I try to include the name in bug reports but I'm sure we aren't consistent.  If we did something with this data though maybe that would give us some incentive to be more vigilant?

My thinking for this column was to provoke exactly this conversation...  So thanks!

Reply:
Actually, I've been surprised, I think we often are given the exact name in the title. I'll pay more attention to this too.